agent/consul/server.go

```diff
@@ -370,9 +370,9 @@ type Server struct {
 
 	// peerStreamServer is a server used to handle peering streams from external clusters.
 	peerStreamServer *peerstream.Server
+
 	// peeringServer handles peering RPC requests internal to this cluster, like generating peering tokens.
-	peeringServer     *peering.Server
-	peerStreamTracker *peerstream.Tracker
```

Comment:
`peerStreamTracker` existed in a weird circular reference with `peerStreamServer`. I removed it from `Server` and it is now contained within `peerStreamServer` which is a narrower and more appropriate context for it to live in.